### PR TITLE
[ONNX] QDQ uint16 support

### PR DESCRIPTION
**Pull request by gkrivor**
### Details:
 - Verification
 - Adding support for int16/uint16 in ConvertQuantizeDequantize transformation

### Tickets:
 - CVS-126167

### CODE REVIEW

File: src/common/transformations/src/transformations/common_optimizations/convert_quantize_dequantize.cpp [hunk 1]
```diff
@@ -135,6 +138,14 @@ ov::pass::ConvertQuantizeDequantize::ConvertQuantizeDequantize() {
             if (out_low_val != 0 || out_high_val != 255)
                 return false;
             break;
+        case element::Type_t::i16:
+            if (out_low_val != -32768 || out_high_val != 32767)
+                return false;
+            break;
+        case element::Type_t::u16:
+            if (out_low_val != 0 || out_high_val != 65536)
```
Comment (slyalin) [inline src/common/transformations/src/transformations/common_optimizations/convert_quantize_dequantize.cpp:146]: After fixing in QuantizeLinear decomposition, it should be reverted back to 65535.

File: src/frontends/onnx/frontend/src/op/quantize_linear.cpp [hunk 2]
```diff
@@ -65,12 +67,26 @@ std::tuple<std::shared_ptr<ngraph::Node>, std::shared_ptr<ngraph::Node>> get_out
...
         output_low = std::make_shared<default_opset::Constant>(data_type, Shape{1}, 0);
         output_high = std::make_shared<default_opset::Constant>(data_type, Shape{1}, 255);
+        break;
+    case element::i16:
+        output_low = std::make_shared<default_opset::Constant>(data_type, Shape{1}, -32768);
+        output_high = std::make_shared<default_opset::Constant>(data_type, Shape{1}, 32767);
+        break;
+    case element::u16:
+        output_low = std::make_shared<default_opset::Constant>(data_type, Shape{1}, 0);
+        output_high = std::make_shared<default_opset::Constant>(data_type, Shape{1}, 65536);
```
[the remark this answers is not in the document]
Comment (gkrivor) [reply src/frontends/onnx/frontend/src/op/quantize_linear.cpp:85]: lets try, maybe my fault

File: src/frontends/onnx/frontend/src/ops_bridge.cpp [hunk 3]
```diff
@@ -576,6 +576,15 @@ OperatorsBridge::OperatorsBridge() {
...
 
+    register_operator_in_custom_domain("DequantizeLinear",
+                                       VersionRange::since(1),
+                                       op::set_13::dequantize_linear,
+                                       "com.microsoft");
+    register_operator_in_custom_domain("QuantizeLinear",
+                                       VersionRange::since(1),
+                                       op::set_13::quantize_linear,
+                                       "com.microsoft");
+
```
Comment (slyalin) [inline src/frontends/onnx/frontend/src/ops_bridge.cpp:587]: `MICROSOFT_DOMAIN` contains `com.microsoft`. Can we use `REGISTER_OPERATOR_WITH_DOMAIN(MICROSOFT_DOMAIN, "DequantizeLinear", 1, op::set_13::dequantize_linear);` as it was utilized for other Microsoft operators?

File: src/frontends/onnx/frontend/src/op/quantize_linear.cpp [hunk 4]
```diff
@@ -34,10 +34,12 @@ Output<ngraph::Node> get_zero_point(const OutputVector& inputs) {
...
     const auto& y_zero_point_et = y_zero_point.get_element_type();
-    CHECK_VALID_NODE(onnx_node,
-                     y_zero_point_et.is_static() && (y_zero_point_et == element::u8 || y_zero_point_et == element::i8),
-                     "\"y_zero_point\" input data type must be static and of 8-bit "
-                     "integer type.");
+    CHECK_VALID_NODE(
+        onnx_node,
+        y_zero_point_et.is_static() && (y_zero_point_et == element::u8 || y_zero_point_et == element::i8 ||
+                                        y_zero_point_et == element::u16 || y_zero_point_et == element::i16),
+        "\"y_zero_point\" input data type must be static and of 8-bit "
```
Comment (slyalin) [inline src/frontends/onnx/frontend/src/op/quantize_linear.cpp:41]:
```suggestion
        "\"y_zero_point\" input for QuantizeLinear operator must be one of the supported types: u8, i8, u16 or i16"
```

File: src/frontends/onnx/frontend/src/op/dequantize_linear.cpp [hunk 5]
```diff
@@ -174,10 +174,14 @@ OutputVector dequantize_linear(const Node& node) {
...
-    if (scale.get_partial_shape().rank().is_static() && scale.get_partial_shape().rank().get_length() == 0) {
-        if (!zero_point || (zero_point->get_output_partial_shape(0).rank().is_static() &&
-                            zero_point->get_output_partial_shape(0).rank().get_length() == 0)) {
+    if (scale_rank.is_static() && (scale_rank.get_length() == 0 || scale_rank.get_length() == 1)) {
+        int64_t zero_point_rank = -1;
+        if (zero_point && zero_point->get_output_partial_shape(0).rank().is_static()) {
+            zero_point_rank = zero_point->get_output_partial_shape(0).rank().get_length();
+        }
+        if (!zero_point || zero_point_rank == 0 || zero_point_rank == 1) {
             return set_1::dequantize_linear(node);
```
Comment (slyalin) [inline src/frontends/onnx/frontend/src/op/dequantize_linear.cpp:185]: This is not correct: `set1::dequantize_linear` cannot work correctly with scale/zero_point with more than one element, there is a condition that will fail in that code. Only 13-th version can deal with that.

Why the code below doesn't work?

**Review by mateusztabaka — APPROVED** [review]
Transformation part LGTM, can you add some tests for ConvertQuantizeDequantize?

**Review by itikhono — APPROVED** [review]

File: src/frontends/onnx/frontend/src/ops_bridge.cpp (hunk 3 above)
Comment (gkrivor) [reply src/frontends/onnx/frontend/src/ops_bridge.cpp:587]: Nop, original macro will produce "op::set_1::op::set_13::quantize_linear"

File: src/frontends/onnx/frontend/src/op/quantize_linear.cpp (hunk 4 above)
Comment (gkrivor) [reply src/frontends/onnx/frontend/src/op/quantize_linear.cpp:41]: applied

**Pull request merged**